python/pylibraft/pylibraft/distance/pairwise_distance.pyx

```diff
@@ -132,6 +142,12 @@ def distance(X, Y, dists, metric="euclidean"):
     y_dt = np.dtype(y_cai["typestr"])
     d_dt = np.dtype(dists_cai["typestr"])
 
+    x_c_contiguous = "strides" not in x_cai or x_cai["strides"] is None
+    y_c_contiguous = "strides" not in y_cai or y_cai["strides"] is None
+
+    if x_c_contiguous != y_c_contiguous:
+        raise ValueError("Inputs must have matching strides")
```

Comment:
I think this check works 99% of the time because (almost?) all NumPy/CuPy functions that create C contiguous arrays make the strides be `None`, but techically speaking the array interface for a C contiguous array can have the valid strides as opposed to `None` and this check would fail. I think it would be good to be resilient to that case, though it could be a follow up since I think the common case should be covered like this